rust/rsc/src/rsc_tool/main.rs

```diff
@@ -65,55 +36,133 @@ async fn list_api_keys(_: ListKeysOpts, conn: &DatabaseConnection) -> Result<(),
-    // First find the key, because its a very strange user expierence to see
-    // "key X was successfully removed" when it never existed in the first place.
-    // It means that you can never be quite sure you acomplished what you thought
-    // you did and have to use list-keys to check. This also lets us output
-    // the description before commiting.
-    let Some(result) = api_key::Entity::find()
-        .filter(api_key::Column::Key.eq(&key))
-        .one(conn)
-        .await?
-    else {
-        println!("{} is not a valid key", key);
+    let uuid = Uuid::parse_str(id)?;
+    let Some(key) = database::read_api_key(db, uuid).await? else {
+        println!("{} is not a valid key", id);
         std::process::exit(2);
     };
 
     // We only want to prompt the user if the user can type into the terminal
     if std::io::stdin().is_terminal() {
         let should_delete = Confirm::new("Are you sure you want to delete this key?")
             .with_default(false)
-            .with_help_message(format!("key = {}, desc = {:?}", key, result.desc).as_str())
+            .with_help_message(format!("key = {}, desc = {:?}", key.key, key.desc).as_str())
             .prompt()?;
 
         if !should_delete {
-            println!("Aborting key removal");
+            println!("Aborting removal");
             return Ok(());
         }
     }
 
     // Ok now that we're really sure we want to delete this key
-    tracing::info!("Deleting key {}", &key);
-    api_key::Entity::delete_many()
-        .filter(api_key::Column::Key.eq(&key))
-        .exec(conn)
-        .await?;
-
-    // If we haven't returned already then log it and tell the user
-    tracing::info!("Key {} deleted", &key);
-    println!("Key {} was successfully removed", key);
+    database::delete_api_key(db, key).await?;
+
+    println!("Key {} was successfully removed", id);
+    Ok(())
+}
+
+async fn add_local_blob_store(
+    opts: AddLocalBlobStoreOpts,
+    db: &DatabaseConnection,
+) -> Result<(), Box<dyn std::error::Error>> {
+    tokio::fs::create_dir_all(opts.root.clone()).await?;
+    let store = database::create_local_blob_store(db, opts.root).await?;
+    println!("Created local blob store: {}", store.id);
+    Ok(())
+}
+
+async fn list_local_blob_stores(db: &DatabaseConnection) -> Result<(), DbErr> {
+    let mut stores: Vec<_> = database::read_local_blob_stores(db)
+        .await?
+        .into_iter()
+        .map(|x| vec![format!("{}", x.id), x.root])
+        .collect();
+
+    let headers = vec!["Id".into(), "Root".into()];
+    stores.insert(0, headers);
+
+    table::print_table(stores);
+
+    Ok(())
+}
+
+async fn remove_local_blob_store(
+    id: &String,
+    db: &DatabaseConnection,
+) -> Result<(), Box<dyn std::error::Error>> {
+    let uuid = Uuid::parse_str(id)?;
+    let Some(store) = database::read_local_blob_store(db, uuid).await? else {
+        println!("{} is not a valid local blob store", id);
```

Comment:
`uuid` here isn't a string its a `Uuid`. The reason for printing `id` is because its the string representation.

`Uuid::parse_str(id)?` will bubble up when `id` isn't a "UUID like string" with an error something like "cannot parse id into UUID"

The wake analogous code is 

```
def foo (id: String) =
    require Pass uuid = id | strToUuid
    require Pass store = readLocalBlobStore uuid
    else failWithError "{id} is not a valid local blob store"
    ...
```